OpenSim/Moco/MocoGoal/MocoContactImpulseTrackingGoal.h

```diff
@@ -0,0 +1,273 @@
+    applied in whatever frame the reference data is expressed in based on the 
+    provided ExternalLoads in each contact group.
+    Adding a scale factor to a MocoImpulseTrackingGoal.
+    @code
+    auto* contactImpulseTrackingGoal = problem.addGoal<MocoContactImpulseTrackingGoal>();
+    ...
+    contactImpulseTrackingGoal->addScaleFactor(
+            'RightGRF_vertical_scale_factor', 'Right_GRF', 1, {0.5, 2.0});
+    @endcode
+    @ingroup mocogoal */
+    class OSIMMOCO_API MocoContactImpulseTrackingGoal : public MocoGoal {
+        OpenSim_DECLARE_CONCRETE_OBJECT(MocoContactImpulseTrackingGoal, MocoGoal);
+
+    public:
+        MocoContactImpulseTrackingGoal() { constructProperties(); }
+        MocoContactImpulseTrackingGoal(std::string name) : MocoGoal(std::move(name)) {
+            constructProperties();
+        }
+        MocoContactImpulseTrackingGoal(std::string name, double weight)
+            : MocoGoal(std::move(name), weight) {
+            constructProperties();
+        }
+
+        /// Set the ExternalLoads as an XML file. This clears the ExternalLoads
+        /// provided as an object, if one exists.
+        void setExternalLoadsFile(const std::string& extLoadsFile);
+        /// Set the ExternalLoads as an object. This clears the ExternalLoads
+        /// XML file, if provided.
+        void setExternalLoads(const ExternalLoads& extLoads);
+
+        /// Add a single group of contact forces whose sum should track the ground 
+        /// reaction force impulse data from a single axis of a single ExternalForce. 
+        /// The externalForceName should be the name of an ExternalForce object
+        /// in the ExternalLoads.
+        void addContactGroup(
+            const std::vector<std::string>& contactForcePaths,
+            const std::string& externalForceName) {
+            append_contact_groups(MocoContactImpulseTrackingGoalGroup(
+                contactForcePaths, externalForceName));
+        }
+        /// Add a single group of contact forces whose sum should track the ground
+        /// reaction force impulse data from a single axis of a single ExternalForce.
+        /// If the contact force elements associated with a single ExternalForce are
+        /// distributed across multiple bodies use this function instead of the
+        /// easier-to-use addContactGroup(), and set the group's
+        /// alternative_frame_paths property accordingly. See
+        /// MocoContactImpulseTrackingGoalGroup for more information.
+        void addContactGroup(MocoContactImpulseTrackingGoalGroup group) {
+            append_contact_groups(std::move(group));
+        }
+
+        /// Set the ground reaction force contact impulse 'axis' to be tracked
+        /// (X = 0, Y = 1, Z = 2), where 'axis' refers to the dimension/direction of
+        /// the ground reaction force contact impulse, in the ground frame, to be
+        /// tracked.
+        void setImpulseAxis(int ImpulseAxis) {
```

Comment:
Small nitpick, but for camelcase variable names, we usually don't capitalize the first letter, i.e., `impulseAxis`.